### DISCUSSION TITLE
Building from source on MacOS - env variables

### DISCUSSION THREAD

**Discussion by JCGoran** [pypy/pypy#4943] [Q&A]
I am trying to build pypy from source using a brew-installed pypy following [the guide](https://doc.pypy.org/en/latest/build.html) on an ARM-based MacOS.
At one step of the installation process (i.e. running `pypy ../../rpython/bin/rpython --opt=jit`), I see:
```plaintext
[platform:WARNING] ld: warning: ignoring file '/usr/local/Cellar/gettext/0.22.4/lib/libintl.8.dylib': found architecture 'x86_64', required architecture 'arm64'
```
which is reasonable since I have both brew for arm64 and brew for x86_64 installed, and I installed `gettext` on both archs.
I thought that this wouldn't be a big deal, but trying to run the `pypy-c` executable fails with missing gettext symbols:
```plaintext
$ ./pypy-c
dyld[85716]: symbol not found in flat namespace '_libintl_bind_textdomain_codeset'
```
My question is, how do I specify the `LDFLAGS` (or whatever env variables pypy installer is looking at) so it finds the right gettext library? I tried setting `LDFLAGS="-L/opt/homebrew/Cellar/gettext/<version>/lib ${LDFLAGS}"`, but that didn't work.

For reference, on ARM-based MacOS, brew installs everything under `/opt/homebrew` instead of `/usr/local`.

**Comment by mattip** [comment]
You should activaye the arm64 brew for an arm64 build, use a arm64 pypy2.7 for translation, and make sure you have not activated the x86_64 brew in that shell.

**Comment by mattip** [comment]
And make sure pkg-config returns the correct gettext

**Comment by JCGoran** [comment]
I managed to get it to work by unlinking the x86_64 version of `gettext` by running `brew unlink gettext` in a Rosetta 2 (x86_64) terminal before compiling. Then the warning message does not appear, and the built `pypy-c` binary runs as expected.
